src/lib.rs

```diff
@@ -97,3 +99,9 @@ pub trait GenFn4<T>: Fn(&PermutationTable, &Point4<T>) -> T {}
 impl<T, F> GenFn2<T> for F where F: Fn(&PermutationTable, &Point2<T>) -> T, {}
 impl<T, F> GenFn3<T> for F where F: Fn(&PermutationTable, &Point3<T>) -> T, {}
 impl<T, F> GenFn4<T> for F where F: Fn(&PermutationTable, &Point4<T>) -> T, {}
+
+pub trait Module<T> {
+    type Output;
+
+    fn get(&self, point: T) -> Self::Output;
+}
```

Comment:
If we add this:

```rust
impl<'a, T, M: Module<T>> Module<T> for &'a M {
    type Output = M::Output;

    #[inline]
    fn get(&self, point: T) -> M::Output {
        M::get(*self, point)
    }
}
```

`Select` wouldn't have to specify whether its modules are taken by value or by ref.